component/s3storage/config.go

```diff
@@ -82,7 +82,9 @@ func ParseAndValidateConfig(s3 *S3Storage, opt Options) error {
 
 	// Part size must be at least 5 MB and smaller than 5GB. Otherwise, set to default.
 	if opt.PartSizeMb < 5 || opt.PartSizeMb > MaxPartSizeMb {
-		log.Warn("ParseAndValidateConfig : Part size must be between 5MB and 5GB. Default to default size")
+		if opt.PartSizeMb != 0 {
```

Comment:
why is opt.PartSizeMb being 0 important to check?

Reply:
This code is meant to log a warning when the user sets `PartSizeMb` outside of the range we allow.
But when `PartSizeMb` is zero, that is supposed to mean that the user didn't set it at all, and they expected us to set it to default. In that case, nothing unexpected is happening, so I figured we shouldn't log a warning.
There is a slight chance that the user explicitly sets `PartSizeMb` to zero, but I'm fine with not logging a warning in that case, since setting a numeric value which needs to be nonzero to zero is a kind of standard way of saying "give me the default value."